go/tasks/plugins/array/k8s/task.go

```diff
@@ -0,0 +1,231 @@
+		actualPhase, err = array.CheckTaskOutput(ctx, dataStore, outputPrefix, baseOutputDataSandbox, t.ChildIdx, originalIdx)
+		if err != nil {
+			return MonitorError, err
+		}
+	}
+
+	t.NewArrayStatus.Detailed.SetItem(t.ChildIdx, bitarray.Item(actualPhase))
+	t.NewArrayStatus.Summary.Inc(actualPhase)
+
+	return MonitorSuccess, nil
+}
+
+func (t Task) Abort(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) error {
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       PodKind,
+			APIVersion: metav1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: tCtx.TaskExecutionMetadata().GetNamespace(),
+		},
+	}
+
+	err := kubeClient.GetClient().Delete(ctx, pod)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+
+			return nil
+		}
+		return err
+	}
+
+	return nil
+
+}
+
+func (t Task) Finalize(ctx context.Context, tCtx core.TaskExecutionContext, kubeClient core.KubeClient) error {
+	indexStr := strconv.Itoa(t.ChildIdx)
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+
+	// Deallocate Resource
+	err := deallocateResource(ctx, tCtx, t.Config, t.ChildIdx)
+	if err != nil {
+		logger.Errorf(ctx, "Error releasing allocation token [%s] in Finalize [%s]", podName, err)
+		return err
+	}
+
+	return nil
+
+}
+
+func allocateResource(ctx context.Context, tCtx core.TaskExecutionContext, config *Config, podName string) (core.AllocationStatus, error) {
+	if !IsResourceConfigSet(config.ResourceConfig) {
+		return core.AllocationStatusGranted, nil
+	}
+
+	resourceNamespace := core.ResourceNamespace(config.ResourceConfig.PrimaryLabel)
+	resourceConstraintSpec := core.ResourceConstraintsSpec{
+		ProjectScopeResourceConstraint:   nil,
+		NamespaceScopeResourceConstraint: nil,
+	}
+
+	allocationStatus, err := tCtx.ResourceManager().AllocateResource(ctx, resourceNamespace, podName, resourceConstraintSpec)
+	if err != nil {
+		logger.Errorf(ctx, "Resource manager failed for TaskExecId [%s] token [%s]. error %s",
+			tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetID(), podName, err)
+		return core.AllocationUndefined, err
+	}
+
+	logger.Infof(ctx, "Allocation result for [%s] is [%s]", podName, allocationStatus)
+	return allocationStatus, nil
+}
+
+func deallocateResource(ctx context.Context, tCtx core.TaskExecutionContext, config *Config, childIdx int) error {
+	if !IsResourceConfigSet(config.ResourceConfig) {
+		return nil
+	}
+	indexStr := strconv.Itoa((childIdx))
+	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
+	resourceNamespace := core.ResourceNamespace(config.ResourceConfig.PrimaryLabel)
+
+	err := tCtx.ResourceManager().ReleaseResource(ctx, resourceNamespace, podName)
```

Comment:
is not found handled here?

Reply:
Yup. I checked and confirmed logs. Will confirm with resource manager API as well.